src/flexible_type/flexible_type_converter.hpp

```diff
@@ -10,657 +10,843 @@
+  template <typename T> static constexpr bool any_match(
+      typename std::enable_if<!ft_converter<idx>::template matches<typename base_type<T>::type>()>::type* = NULL) {
+    return ft_resolver<idx - 1>::template any_match<T>();
+  }
+
+  ////////////////////////////////////////////////////////////////////////////////
+  // Does this idx, and none before this one, match?
+
+  template <typename T> static constexpr bool matches() {
+      return (ft_converter<idx>::template matches<typename base_type<T>::type>()
+              && !ft_resolver<idx - 1>::template any_match<T>());
+  }
+
+  ////////////////////////////////////////////////////////////////////////////////
+  // Does this idx, and none before this one, match?
+
+  template <typename T>
+  static void get(T& t, const flexible_type& v,
+                  typename std::enable_if<matches<T>()>::type* = NULL) {
+    ft_converter<idx>::get(t, v);
+  }
+
+  template <typename T>
+  static void get(T& t, const flexible_type& v,
+                  typename std::enable_if<!matches<T>()>::type* = NULL) {
+    ft_resolver<idx - 1>::get(t, v);
+  }
+
+  template <typename T>
+  static void set(flexible_type& t, T&& v,
+                  typename std::enable_if<matches<T>()>::type* = NULL) {
+    ft_converter<idx>::set(t, std::forward<T>(v));
   }
-  flexible_type set(const T& val) {
-    return flexible_type(static_cast<flex_int>(val));
+
+  template <typename T>
+  static void set(flexible_type& t, T&& v,
+                  typename std::enable_if<!matches<T>()>::type* = NULL) {
+    ft_resolver<idx - 1>::set(t, std::forward<T>(v));
   }
 };
 
+// The base case in which any_match finally returns false
+template <> struct ft_resolver<0> {
+  template <typename T> static constexpr bool any_match() { return false; }
+  template <typename T> static constexpr bool matches() { return false; }
+  template <typename... T> static void set(T...) {}
+  template <typename... T> static void get(T...) {}
+};
+
+}
 
 /**
  * is_flexible_type_convertible<T>::value is true if T can be converted
  * to and from a flexible_type via flexible_type_converter<T>.
  */
 template <typename T>
 struct is_flexible_type_convertible {
-  static constexpr bool value = flexible_type_converter<T>::value;
-  typedef boost::mpl::bool_<value> type;
+  static constexpr bool value = flexible_type_internals::ft_resolver<
+    flexible_type_internals::NUM_CONVERTER_STRUCTS-1>::template any_match<typename base_type<T>::type>();
 };
 
-/**
- * all_flexible_type_convertible<Args...>::value is true if every type in 
- * Args can be converted to and from a flexible_type via 
- * flexible_type_converter<T>.
+template <typename T> GL_HOT_INLINE_FLATTEN
+static void convert_from_flexible_type(T& t, const flexible_type& f) {
+  static_assert(is_flexible_type_convertible<T>::value, "Type not convertable from flexible_type.");
+
+  flexible_type_internals::ft_resolver<flexible_type_internals::NUM_CONVERTER_STRUCTS-1>::get(t, f);
```

Comment:
@znation get is called here.

Reply:
Seems a bit too magical to me, but I'll go with it. 👍